### PR TITLE
Documentation revamp + Fetch Migration

### PR DESCRIPTION
**Pull request by kartg**
### Description

This PR cleans up some stale README file and includes instructions for Fetch Migration into the documentation. It also adds a detailed, step-by-step guide on deploying to AWS using CDK and Copilot.

Some other minor updates included in this PR:
* Since Data Prepper 2.5.0 with the OpenSearch source plugin [has now been released](https://opensearch.org/blog/Announcing-Data-Prepper-2.5.0/), the Fetch Migration Dockerfile now uses this as the base image rather than a snapshot
* The Python dependency versions for Fetch Migration have been updated
* Updated the `runTestBenchmarks.sh` script to allow a `no-ssl` flag
* Minor bugfixes in `fetch_orchestrator.py` identified by @okhasawn:
  * Fixing an error in the ArgParse name for the Data Prepper endpoint
  * Updating the mechanism to fetch the INLINE_PIPELINE environment variable to avoid KeyError. 
* Increased the default CPU and Memory specification for the Fetch Migration container

### Issues Resolved
N/A

### Testing
N/A

### Check List
- [ ] New functionality includes testing
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: FetchMigration/README.md [hunk 1]
```diff
@@ -1,43 +1,38 @@
...
-* Only supports ElasticSearch and OpenSearch endpoints for source and target
-* Only supports basic auth
-* Type mappings for legacy indices are not handled
-* Index templates and index aliases are not copied
-* Index health is not validated after creation
-
-## Usage
+* Fetch Migration runs as a single instance and does not support vertical scaling or data slicing
+* The tool does not support customizing the list of indices included for migration
+* Metadata migration only supports basic auth
```
Comment (sumobrian) [inline FetchMigration/README.md:29]: Isn't this a problem for Serverless?

File: FetchMigration/README.md [hunk 2]
```diff
@@ -1,43 +1,38 @@
...
+* A "metadata migration" module that handles metadata comparison between the source and target clusters. 
+This can output a human-readable report as well as a Data Prepper pipeline `yaml` file.
+* A "migration monitor" module that monitors the progress of the migration and shuts down the Data Prepper pipeline 
+once the target document count has been reached
+* An "orchestrator" module that sequences these steps as a workflow and manages the kick-off of the Data Prepper 
+process between them.
 
-### Reporting
-
-If `--report` is specified, the tool prints all processed indices organized into 3 buckets:
```
Comment (okhasawn) [inline FetchMigration/README.md:23]: Is this no longer the case?

File: FetchMigration/README.md [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
-* Only supports basic auth
-* Type mappings for legacy indices are not handled
-* Index templates and index aliases are not copied
-* Index health is not validated after creation
-
-## Usage
+* Fetch Migration runs as a single instance and does not support vertical scaling or data slicing
+* The tool does not support customizing the list of indices included for migration
+* Metadata migration only supports basic auth
+* The migration does not filter out `red` indices
```
Comment (okhasawn) [inline FetchMigration/README.md:30]: Why is this a limitation? what if at the time of migration, that index was red, but the user still wanted that index to be migrated?

Comment (kartg) [reply FetchMigration/README.md:30]: a `red` index means one or more primary shards are unassigned. For indices that were previously non-red, this means that some portion of the index is not readable. Thus, even though migrating such an index would go through, the data on the target index would be incomplete.

File: FetchMigration/README.md (hunk 2 above)
Comment (kartg) [reply FetchMigration/README.md:23]: Yes. It is turned on by default in `fetch_orchestrator.py` to allow for meaningful human-readable log output:

https://github.com/kartg/opensearch-migrations/blob/e9501a071c4bf7be7006d8e91e522d57b65ae920/FetchMigration/python/fetch_orchestrator.py#L20

File: deployment/cdk/opensearch-service-migration/README.md [hunk 4]
```diff
@@ -57,12 +57,12 @@ This is the core required stack of this CDK which is responsible for deploying t
...
 This optional stack will be used when the Domain is configured to be placed inside a VPC and will contain resources related to the networking of this VPC such as Security Groups and Subnets. It has a dependency on the Domain stack.
 
-#### Historical Capture Stack (OSServiceHistoricalCDKStack-STAGE-REGION)
-This optional stack sets up a ECS cluster to host/run Fetch Migration tasks for historical data migration. It has dependencies on both the Domain and Network stacks.
-
 #### Migration Assistance Stack (OSServiceMigrationCDKStack-STAGE-REGION)
 This optional stack is used to house the migration assistance resources which are in the process of being developed to assist in migrating to an OpenSearch domain. It has dependencies on both the Domain and Network stacks.
 
+#### Fetch Migration Stack (OSServiceHistoricalCDKStack-STAGE-REGION)
+This optional stack sets up a ECS cluster to host/run Fetch Migration tasks for data backfill / historical data migration. It has dependencies on the Domain, Network and Migration Assistance stacks.
```
Comment (sumobrian) [inline deployment/cdk/opensearch-service-migration/README.md:64]: nit: 
"an ECS cluster"

nit: inconsistency on spacing around '/' (host/run and backfill / historical)

File: FetchMigration/README.md [hunk 5]
```diff
@@ -67,42 +60,46 @@ docker build -t fetch-migration .
...
 
 ```shell
-python -m unittest
+python -m coverage run -m unittest
 ```
 
-### Coverage
-
-Code coverage metrics can be generated by first running unit tests using _coverage run_:
+_Code coverage_ metrics can be generated after a unit-test run. A report can either be printed on the command line:
```
Comment (sumobrian) [inline FetchMigration/README.md:92]: nit: not sure why the ### Coverage (h3) was removed.

**Review by sumobrian — APPROVED** [review]
Minor nitpick changes.

File: FetchMigration/README.md (hunk 5 above)
Comment (kartg) [reply FetchMigration/README.md:92]: Good find, that was a mistake. Fixed now.

File: deployment/cdk/opensearch-service-migration/README.md (hunk 4 above)
Comment (kartg) [reply deployment/cdk/opensearch-service-migration/README.md:64]: Fixed

**Pull request merged**